src/transaction.rs

```diff
@@ -0,0 +1,254 @@
+            if val.is_null() {
+                Ok(None)
+            } else {
+                Ok(Some(DBVector::from_c(val, val_len)))
+            }
+        }
+    }
+
+    /// Read a key inside a transaction
+    ///
+    /// ReadOptions: default
+    pub fn get_cf(&self, cf: ColumnFamily, key: &[u8]) -> Result<Option<DBVector>, Error> {
+        self.get_cf_opt(cf, key, ReadOptions::default())
+    }
+
+    /// Insert a value into the database under the given key.
+    ///
+    /// ColumnFamilyHandle: default
+    pub fn put(&self, key: &[u8], value: &[u8]) -> Result<(), Error> {
+        unsafe {
+            ffi_try!(ffi::rocksdb_transaction_put(
+                self.inner,
+                key.as_ptr() as *const c_char,
+                key.len() as size_t,
+                value.as_ptr() as *const c_char,
+                value.len() as size_t,
+            ));
+            Ok(())
+        }
+    }
+
+    /// Insert a value into the database under the given key.
+    pub fn put_cf(&self, cf: ColumnFamily, key: &[u8], value: &[u8]) -> Result<(), Error> {
+        unsafe {
+            ffi_try!(ffi::rocksdb_transaction_put_cf(
+                self.inner,
+                cf.inner,
+                key.as_ptr() as *const c_char,
+                key.len() as size_t,
+                value.as_ptr() as *const c_char,
+                value.len() as size_t,
+            ));
+            Ok(())
+        }
+    }
+
+    /// Merge a key inside a transaction
+    pub fn merge(&self, key: &[u8], value: &[u8]) -> Result<(), Error> {
+        unsafe {
+            ffi_try!(ffi::rocksdb_transaction_merge(
+                self.inner,
+                key.as_ptr() as *const c_char,
+                key.len() as size_t,
+                value.as_ptr() as *const c_char,
+                value.len() as size_t,
+            ));
+            Ok(())
+        }
+    }
+
+    /// Transaction rollback
+    pub fn rollback(&self) -> Result<(), Error> {
+        unsafe { ffi_try!(ffi::rocksdb_transaction_rollback(self.inner,)) }
+        Ok(())
+    }
+
+    /// Transaction rollback to savepoint
+    pub fn rollback_to_savepoint(&self) -> Result<(), Error> {
+        unsafe { ffi_try!(ffi::rocksdb_transaction_rollback_to_savepoint(self.inner,)) }
+        Ok(())
+    }
+
+    /// Set savepoint for transaction
+    pub fn set_savepoint(&self) {
+        unsafe { ffi::rocksdb_transaction_set_savepoint(self.inner) }
+    }
+}
+
+impl<'a> Drop for Transaction<'a> {
+    fn drop(&mut self) {
+        self.snapshot = None;
+        unsafe {
+            ffi::rocksdb_transaction_destroy(self.inner);
+        }
+    }
+}
+
+impl<'a> CreateIter for Transaction<'a> {
+    fn raw_iter(&self, readopts: &ReadOptions) -> DBRawIterator {
+        unsafe {
+            DBRawIterator {
+                inner: ffi::rocksdb_transaction_create_iterator(self.inner, readopts.inner),
+                db: PhantomData,
+            }
+        }
+    }
+
+    fn raw_iter_cf(
```

Comment:
Can `cf_handle` be a reference, eg: `cf_handle: &ColumnFamily`?

Reply:
ColumnFamily impl Copy

Reply:
True, but I'd rather not depend on that.  Frankly, having `ColumnFamily` impl `Copy` when it contains a pointer is a bit of a code-smell to me.  If possible, I'd like to see that dropped.

Reply:
agree